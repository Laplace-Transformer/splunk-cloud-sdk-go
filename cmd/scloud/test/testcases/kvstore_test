kvstore create-index --name myindex --fields "[{\"direction\":-1,\"field\":\"integ_testField1\"}]" --collection gomod1582754557.gointegdskvidx_1582754557
kvstore delete-index --collection gomod1582754557.gointegdskvidx_1582754557 --index myindex
kvstore list-indexes --collection gomod1582754557.gointegdskvidx_1582754557
kvstore ping
kvstore insert-record --collection gomod1582754557.gointegdskvidx_1582754557 --body "{\"capacity_gb\": 8}"
kvstore list-records --collection gomod1582754557.gointegdskvidx_1582754557
kvstore insert-records --collection gomod1582754557.gointegdskvidx_1582754557 --body "[{\"capacity_gb\": 8},{\"capacity_gb\": 7}]"
kvstore put-record --collection gomod1582754557.gointegdskvidx_1582754557 --key "5e58525673875400011ffb13"  --body "{\"capacity_gb\": 8}"
kvstore query-records --collection gomod1582754557.gointegdskvidx_1582754557
kvstore get-record-by-key --collection gomod1582754557.gointegdskvidx_1582754557 --key "5e58525673875400011ffb13"
kvstore delete-record-by-key --collection gomod1582754557.gointegdskvidx_1582754557 --key "5e58525673875400011ffb13"
kvstore delete-records --collection gomod1582754557.gointegdskvidx_1582754557


