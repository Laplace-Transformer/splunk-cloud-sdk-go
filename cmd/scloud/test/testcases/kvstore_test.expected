#testcase: kvstore create-index --name myindex --fields "[{\"direction\":-1,\"field\":\"integ_testField1\"}]" --collection gomod1582754557.gointegdskvidx_1582754557
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557/indexes
REQUEST BODY:{{"fields":[{"direction":-1,"field":"integ_testField1"}],"name":"myindex"}}

#testcase: kvstore delete-index --collection gomod1582754557.gointegdskvidx_1582754557 --index myindex
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557/indexes/myindex
REQUEST BODY:<nil>

#testcase: kvstore list-indexes --collection gomod1582754557.gointegdskvidx_1582754557
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557/indexes
REQUEST BODY:<nil>

#testcase: kvstore ping
REQUEST URL:kvstore/v1beta1/ping
REQUEST BODY:<nil>

#testcase: kvstore insert-record --collection gomod1582754557.gointegdskvidx_1582754557 --body "{\"capacity_gb\": 8}"
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557
REQUEST BODY:{{"capacity_gb":8}}

#testcase: kvstore list-records --collection gomod1582754557.gointegdskvidx_1582754557
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557
REQUEST BODY:<nil>

#testcase: kvstore insert-records --collection gomod1582754557.gointegdskvidx_1582754557 --body "[{\"capacity_gb\": 8},{\"capacity_gb\": 7}]"
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557/batch
REQUEST BODY:{[{"capacity_gb":8},{"capacity_gb":7}]}

#testcase: kvstore put-record --collection gomod1582754557.gointegdskvidx_1582754557 --key "5e58525673875400011ffb13"  --body "{\"capacity_gb\": 8}"
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557/records/%225e58525673875400011ffb13%22
REQUEST BODY:{{"capacity_gb":8}}

#testcase: kvstore query-records --collection gomod1582754557.gointegdskvidx_1582754557
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557/query
REQUEST BODY:<nil>

#testcase: kvstore get-record-by-key --collection gomod1582754557.gointegdskvidx_1582754557 --k
enable testhook-dryrun
required flag(s) "key" not set

#testcase: kvstore delete-record-by-key --collection gomod1582754557.gointegdskvidx_1582754557 --key "5e58525673875400011ffb13"
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557/records/5e58525673875400011ffb13
REQUEST BODY:<nil>

#testcase: kvstore delete-records --collection gomod1582754557.gointegdskvidx_1582754557
REQUEST URL:kvstore/v1beta1/collections/gomod1582754557.gointegdskvidx_1582754557/query
REQUEST BODY:<nil>

